beacon-chain/core/blocks/withdrawals.go

```diff
@@ -148,31 +148,50 @@ func ValidateBLSToExecutionChange(st state.ReadOnlyBeaconState, signed *ethpb.Si
 //	        next_validator_index = ValidatorIndex(next_index % len(state.validators))
 //	        state.next_withdrawal_validator_index = next_validator_index
 func ProcessWithdrawals(st state.BeaconState, executionData interfaces.ExecutionData) (state.BeaconState, error) {
+	if st.Version() >= version.EPBS {
+		IsParentBlockFull, err := st.IsParentBlockFull()
+		if err != nil {
+			return nil, errors.Wrap(err, "could not check if parent block is full")
+		}
+
+		if !IsParentBlockFull {
+			return nil, nil
+		}
+	}
+
 	expectedWithdrawals, partialWithdrawalsCount, err := st.ExpectedWithdrawals()
 	if err != nil {
 		return nil, errors.Wrap(err, "could not get expected withdrawals")
 	}
 
 	var wdRoot [32]byte
-	if executionData.IsBlinded() {
-		r, err := executionData.WithdrawalsRoot()
+	if st.Version() >= version.EPBS {
+		r, err := st.LastWithdrawalsRoot()
```

Comment:
This is the opposite of the spec:
```python
    withdrawals, partial_withdrawals_count = get_expected_withdrawals(state)
    withdrawals_list = List[Withdrawal, MAX_WITHDRAWALS_PER_PAYLOAD](withdrawals)
    state.latest_withdrawals_root = hash_tree_root(withdrawals_list)
  ```
  We need to get the expected withdrawals from the state and take the root of this, this htr is then set in the state as the latest withdrawals root. You are instead getting the latest withdrawals root from the current state. 
 
 See here for the python implementation
 https://github.com/ethereum/consensus-specs/pull/3854/files#diff-a0aceafe962ba8f0bb107123ceaebc7d54408eb68504355b9ef76376f10e1cbeR452-R454

Reply:
Ah! makes sense actually thks 👍